### PR TITLE
Add command-line flags to run processes outside of k8s

### PR DESCRIPTION
**Pull request by caboteria**
When you're developing, running components at the command line and
having them attach to a cluster might be more convenient than
redeploying them to a cluster over and over.

For the controller, -kubeconfig and -config-ns are needed.  Speakers
need those and -node-name.

For example:

 metallb$ go run ./controller/main.go ./controller/service.go -config-ns metallb-system -kubeconfig $KUBECONFIG

 metallb$ go run ./speaker/main.go ./speaker/*controller.go -config-ns metallb-system -kubeconfig $KUBECONFIG -node-name node0

Thanks for sending a pull request! A few things before we get started:

1. If this is your first time, please read the [contributing guide](https://metallb.universe.tf/community/#contributing)
2. For non-trivial pull requests, please [file an
   issue](https://github.com/google/metallb/issues/new) first, and get
   agreement that the change is a good idea, and a general guideline
   for how it should be implemented, before sending code. Large PRs
   that weren't first discussed and agreed upon in an issue won't be
   accepted.
3. If the PR fixes a particular bug, please include the words "Fixed
   #<issue number>" in the PR text, so that the bug auto-closes when
   the PR is merged.

### CODE REVIEW

**Comment by rata** [comment]
Hi!

Sorry for the quick answer, I'm on my phone.

It's not very well documented here, https://metallb.universe.tf/community/, but basically you you install those things you can deploy locally. You can run all invoke commands, for local development inv dev-env (or alike, don't remember and I'm on my phone so not trivial to check) will do what you want automatically and quite easily (first time will create a cluster, next time you run it will recompile and update metalLB).

Just curious, have you tried that?

**Comment by caboteria** [comment]
Yup, dev-env is a useful tool and it works well.

The changes I'm proposing add another option that benefits developers in a few ways:
 * speed: running "dev-env" with an update takes about 15s because it builds docker images, pushes them into the cluster, undeploys everything in the cluster, and then redeploys everything in the cluster.  Running "go run" externally takes about 2s for the controller to run.
 * flexibility: these changes let you attach a locally-running controller or speaker to *any* cluster that you've got a kubeconfig file for.  This is great for testing setups with special networking config or complex apps.
 * debug: because the local process is running on the host and not inside a docker container inside kind it should be easier to attach a debugger to it (although I haven't tested that yet).

**Comment by russellb** [comment]
This seems like a useful addition to help an alternate dev flow.  I'd use this, as I'm more likely to do dev against a different cluster than the dev env cluster.

Can you add some docs about it to the page @rata mentioned, though?

**Comment by caboteria** [comment]
No problem!

File: website/content/community/_index.md [hunk 1]
```diff
@@ -127,6 +127,19 @@ registries, and so forth. `inv -l` lists the available tasks.
...
+When you're developing, running components at the command line and
+having them attach to a cluster might be more convenient than
+redeploying them to a cluster over and over.
+
+For the controller, the `-kubeconfig` and `-config-ns` command-line flags
+are needed.  Speakers need those and `-node-name`.
+
+For example:
+
+ metallb$ go run ./controller/main.go ./controller/service.go -config-ns metallb-system -kubeconfig $KUBECONFIG
```
Comment (russellb) [inline website/content/community/_index.md:139]: As a future idea, maybe a tasks.py addition of a "run-controller" command would be a little easier.  same with "run-speaker".

**Comment by russellb** [comment]
lgtm, pretty simple change, doesn't get in the way if you don't need it, and I would use this in my personal dev workflow.

**Comment by russellb** [comment]
cc @rata

**Pull request merged**